zerver/lib/user_topics.py

```diff
@@ -236,11 +236,9 @@ def is_muted(recipient_id: int, topic: str) -> bool:
     return is_muted
 
 
-def get_users_muting_topic(stream_id: int, topic_name: str) -> QuerySet[UserProfile]:
-    return UserProfile.objects.select_related("realm").filter(
-        id__in=UserTopic.objects.filter(
-            stream_id=stream_id,
-            visibility_policy=UserTopic.VisibilityPolicy.MUTED,
-            topic_name__iexact=topic_name,
-        ).values("user_profile_id")
-    )
```

Comment:
Is there any specific reason why this query was as complicated earlier as it was? In case there's some optimization that writing this query like this does, we wouldn't want to give up its benefits by simplifying the query.
(Maybe the query was just written badly for no specific reason. We should simplify it in a separate prep commit with the commit message saying so. To be sure, you can use the Git "pickaxe" to find the commit which wrote this query -- hopefully it will have some explanation as to why it is like this.)

Reply:
I think it was trying to just return `UserProfile` objects rather than `UserTopic` objects with a `.user_profile`, and this is one of the simpler constructions for doing that.

Reply:
It was included in e45cebd636327f05b68a2ab841a5c7c6493a130f

Earlier our purpose was fulfilled only with **user_profiles** because we **only** had MUTED as the visibility_policy.
Now when a topic is moved we need `user_profile` and the `visibility_policy` the user has set (for each users who have set some visibility_policy for that topic).

So, either we should return a dictionary `{user_profile : visibility_policy}` or `user_topic` (and fulfill our need via `user_topic.user_profile` + `user_topic.visibility_policy`}

In the next commit (regarding bulk operation), we would need `{visibility_policy : set of user_profiles}`. So, again the updated query will be utilized (only user_profiles doesn't serve our purpose).